core/integration/stubs/broker_stub.go

```diff
@@ -936,6 +936,21 @@ func (b *BrokerStub) VestingStats() []eventspb.VestingStatsUpdated {
 	return stats
 }
 
+func (b *BrokerStub) VolumeDiscountStats() []eventspb.VolumeDiscountStatsUpdated {
+	batch := b.GetBatch(events.VolumeDiscountStatsUpdatedEvent)
+
+	stats := make([]eventspb.VolumeDiscountStatsUpdated, 0, len(batch))
+	for _, event := range batch {
+		switch et := event.(type) {
+		case *events.VolumeDiscountStatsUpdated:
+			stats = append(stats, et.Proto())
```

Comment:
Probably out of scope of this PR, but the `events.Proto()` functions are inconsistent WRT their return values. `Party.Proto()` returns `*proto.Party`, whereas most other types like `RiskFactor.Proto()` return `proto.RiskFactor`. There may be a good reason for this behaviour, but if there isn't, we probably want to make this consistent at some point.

Reply:
Created the issue.